app/frontend/dabates/debates.reducers.js

```diff
@@ -0,0 +1,181 @@
+import { 
+  FETCH_PROPOSALS, 
+  FETCH_PROPOSAL, 
+  UPDATE_PROPOSAL,
+  APPEND_PROPOSALS_PAGE, 
+  VOTE_PROPOSAL,
+  UPDATE_ANSWER,
+  FETCH_RELATED_MEETINGS,
+  FETCH_REFERENCES,
+  FETCH_ACTION_PLANS,
+  HIDE_PROPOSAL,
+  HIDE_PROPOSAL_AUTHOR,
+  FLAG_PROPOSAL,
+  UNFLAG_PROPOSAL,
+} from './proposals.actions';
+
+import { 
+  FOLLOW, 
+  UNFOLLOW, 
+  FETCH_FOLLOW 
+} from '../follows/follows.actions';
+
+import { 
+  FETCH_COMMENTS, 
+  APPEND_COMMENTS_PAGE, 
+  ADD_NEW_COMMENT,
+  FLAG_COMMENT,
+  UNFLAG_COMMENT,
+  UPVOTE_COMMENT,
+  DOWNVOTE_COMMENT,
+  HIDE_COMMENT,
+  HIDE_COMMENT_AUTHOR
+} from '../comments/comments.actions';
+
+import { comments } from  '../comments/comments.reducers';
+
+export const proposals = function (state = [], action) {
+  switch (action.type) {
+    case FETCH_PROPOSALS:
+      return action.payload.data.proposals
+    case APPEND_PROPOSALS_PAGE:
+      return [
+        ...state,
+        ...action.payload.data.proposals
+      ];
+    case VOTE_PROPOSAL:
+      return state.map(p => proposal(p, action));
+  }
+  return state;
+}
+
+export const proposal = function (state = {}, action) {
```

Comment:
Not sure I like this ultra-case duplicated here. Isn't there any cleaner solution?